packer/builder/azure/common/devicelogin.go

```diff
@@ -0,0 +1,244 @@
+package common
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/subscriptions"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/Azure/packer-azure/version"
+	"github.com/mitchellh/go-homedir"
+)
+
+var (
+	// AD app id for docker-machine driver in various Azure realms
```

Comment:
mentions docker-machine. Also, do we need a separate app/client ID/s for packer-azure?